Handle water utility cost tariffs

jmarrec opened the pull request:
Fixes #6168 to clarify the conversion factors in UtilityCost:Tariff, especially when used with a water resource.

- [x] Add one or more conversion factor choices for water to the IDD and getinput.
- [x] Edit the IDD notes to clarify how the default will be selected when the conversion factor is blank.
- [x] Edit the I/O Ref to match the changes and to state what heating value was assumed for the CCF and MCF conversion factors which convert from energy to volume.  Currently, the I/O Ref says a blank in this field will results in conversion factors of 1.0, which is not consistent with current and/or proposed code.
- [x] Related code changes in `GetInputEconomicsTariff`
- [x] Possibly remove what appears to be duplicate logic in `selectTariff`

jmarrec committed:
Added checks to see if the Meter is a water meter or not

jmarrec committed:
Typos and forgot to change the header file. Compiles now.

jmarrec committed:
Add explicit setting to kindMeterNotWater by default. This was already handled in the constructor (set to 0), but is this standard (or better) practice to include it?
Has a TODO

jmarrec committed:
Check for Meter Water type in UpdateUtilityBill instead (should electric check be there too?) and fix the Gtest. Compiles and test passes

nrel-bot-2 commented:
@jmarrec @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@jmarrec @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@jmarrec @lgentile it has been 14 days since this pull request was last updated.

rraustad commented on src/EnergyPlus/EconomicTariff.cc:
```diff
@@ -482,11 +485,67 @@ namespace EconomicTariff {
-				ShowContinueError( cAlphaFieldNames( 3 ) + "=\"" + cAlphaArgs( 3 ) + "\", Defaulting to KWH." );
+
+
+				// TODO: is this needed?
```
TODO? Multiple places. Is there something to do here or not?

jmarrec replied:
In two place to be exact, here and line 4643. I'm not sure whether I need to put this if statement here or there

jmarrec replied:
(And in line 498, comment is explicit). Maybe you're just more used to using github to place comments on each line (tedious here...), I just used TODO (consistently, it colors differently on my text editor) as a comment-in-code feature for the person reviewing the changes, to explain what I've done or which questions I have, hoping to provide clarity. Maybe that was a hit and miss, in which case I'm sorry!

rraustad replied:
I like clarity, just wasn't used to this method. If it had said "For reviewer: is this needed?" I would have got it immediately.  So if tariff.kindWaterMtr is already 0, then this is not needed. BUT, initializing to an explicit is not wrong either. So I think this is easier regarding maintenance than searching for the constructor default.

mjwitte replied:
Since the default varies depending on the resource type, then I'm ok with a warning that says what conversion it defaulted to.  However, we should be adding M3 or some equivalent to the list of predefined choices.  There is also the problem that water could be billed in CCF or MCF, but currently these choices assume natural gas as some (unstated) heating value.  So, perhaps these should be changed to CCFNaturalGas, and MCFNaturalGas, and then add CCFH2O and MCFH20 and M3H20?   Seem messy, but unless we add a new field to declare the resource type, then CCF is too ambiguous.  @JasonGlazer should chime in here.

JasonGlazer replied:
The Tariff is connected to a specific meter that has a resource so strictly it is probably not required but for clarity I think it would make sense to have specific units for the IP water units. Normally we use ft3 or gal for water units but I understand the utilities tend to use CCF. I guess I would recommend we use CCFH20.

mjwitte replied:
@JasonGlazer You make a good point.  The code knows the base units for a given meter, so it can just be smart enough to apply a different factor for CCF on a water meter vs CCF on a gas meter.  And CCF on an electric meter should be an error.  So, this can be fixed without any IDD changes and update the docs.  If we want to go a step further, then we could also add gallons and 1000gallons, but that could wait.

jmarrec replied:
We need to add "m^3" as a choice in the IDD (in which case we might as well add gal and ft^3, perhaps kgal too, CCF is already included). (fyi, Most utilities I've deal with actually use HCF (=CCF) on their bill.)

rraustad commented on src/EnergyPlus/EconomicTariff.cc:
```diff
@@ -482,11 +485,67 @@ namespace EconomicTariff {
+
+
+				// If it's a water meter, default to m^3, otherwise will default to kWh
+				if ( tariff( iInObj ).kindWaterMtr == kindMeterWater ) {
```
Not sure of this but asking anyway. If we know it's a water meter, do we need to tell the user it's m3/s or otherwise it's kWh ?

rraustad replied:
It's here that I don't think this is needed. The user expects m3/s for water, and kWh or other if not water. So I would delete this unless @mjwitte has a reason to keep.

jmarrec replied:
Just to be clear, you're saying to remove the ShowWarning? I just copied the pattern I've seen where it does tell you that it defaulted to kWh. Seems like a valid information I'd like to have (and if I don't like the warning I can modify the object to suppress it). My 2 cents.

rraustad replied:
Does the documentation say a water meter will have units of m3/s ? If it does, then no need to tell the user again. If it doesn't, then the documentation should be updated. Why is it a warning that water meters use m3/s as the units?

mjwitte replied:
The documentation is silent about water.  That will need updates as well.  I'll add a checklist at the top of the pull request.

rraustad commented:
@Myoldmopar @mjwitte I've reviewed and this seems reasonable. @jmarrec please post some results so reviewers have an easier time assessing the impact/change. This way, if someone else does run a test they'll know what to look for.

jmarrec commented:
What do you mean by "post some results"?
Is the Gtest I added not enough?

rraustad commented:
For a reviewer, it's hard to know that a change has corrected something unless they have something to compare. You found this issue by reviewing results of tables, or eio, or something. Can you paste a pic of a table or other result that shows the water meter is now in units of m3/s (a before and after pic would be ideal). Yes, your test confirms this change works, but where in the simulation results would someone find m3/s instead of kWh. I'm lazy, I want to be shown where to look (because I don't always know).

mjwitte commented on src/EnergyPlus/EconomicTariff.cc:
```diff
@@ -4581,9 +4640,14 @@ namespace EconomicTariff {
+			tariff( iTariff ).kindWaterMtr = kindMeterNotWater;
+
+			//determine if this is meter related to electricity, or water
 			if ( tariff( iTariff ).reportMeterIndx != 0 ) {
```
This is more a question for @JasonGlazer .  If the kind of meter has already been set in GetInputEconomicsTariff, why does this logic need to be repeated here?

JasonGlazer replied:
It isn't repeated. kindElectricMtr is not determined during GetInput.

mjwitte replied:
Oh - I see now, the proposed changes include a commented out block in GetInput.  So depending on how this turns out, it might make sense to move it up there.

jmarrec replied:
Can I get a ruling on this please? I think it makes more sense to move it in `GetInputEconomicsTariff`

mjwitte reviewed (commented):
See pull request description [above](https://github.com/NREL/EnergyPlus/pull/6169#issue-234867593) for a checklist of items that should be cleaned up for this.  The proposed solution is helpful, but it still leaves some confusion.  @jmarrec  It's up to you if you want to tackle all of the parts or not.

Myoldmopar commented:
Thanks for looking over all of this @mjwitte @rraustad @JasonGlazer.  @mjwitte noted a checklist of items that still needed to be done for this to go in.  @jmarrec is that something you're able to tackle?  I'd love to get this change in, and I am happy to review the work once complete, but I don't have the bandwidth to tackle making more changes today/this weekend.  Our IDD freeze has technically passed, so unless you are able to make the changes to complete the checklist, I think this will have to wait.

I'll mark it for version "9.0" (which may be 8.9), so it can drop in as soon as we've released 8.8.  If you do get around to doing @mjwitte's checklist of requests today, and it looks good, just ping me and I can try to get this into 8.8.  If you want to simply hand it off and let us finish it up, that's cool too!

However it works out, thanks for all your contributions!

mjwitte commented:
Actually the checklist is out of date. See my comment from Thursday or Friday. We can improve this for 8.8 without an IDD change if so desired.


-------- Original message --------
From: Edwin Lee <notifications@github.com> 
Date: 8/19/17  6:47 AM  (GMT-06:00) 
To: NREL/EnergyPlus <EnergyPlus@noreply.github.com> 
Cc: "Michael J. Witte" <mjwitte@gard.com>, Mention <mention@noreply.github.com> 
Subject: Re: [NREL/EnergyPlus] Handle water utility cost tariffs (#6169) 

Thanks for looking over all of this @mjwitte @rraustad @JasonGlazer.  @mjwitte noted a checklist of items that still needed to be done for this to go in.  @jmarrec is that something you're able to tackle?  I'd love to get this change in, and I am happy to review the work once complete, but I don't have the bandwidth to tackle making more changes today/this weekend.  Our IDD freeze has technically passed, so unless you are able to make the changes to complete the checklist, I think this will have to wait.
I'll mark it for version "9.0" (which may be 8.9), so it can drop in as soon as we've released 8.8.  If you do get around to doing @mjwitte's checklist of requests today, and it looks good, just ping me and I can try to get this into 8.8.  If you want to simply hand it off and let us finish it up, that's cool too!
However it works out, thanks for all your contributions!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/NREL/EnergyPlus","title":"NREL/EnergyPlus","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/NREL/EnergyPlus"}},"updates":{"snippets":[{"icon":"PERSON","message":"@Myoldmopar in #6169: Thanks for looking over all of this @mjwitte @rraustad @JasonGlazer.  @mjwitte noted a checklist of items that still needed to be done for this to go in.  @jmarrec is that something you're able to tackle?  I'd love to get this change in, and I am happy to review the work once complete, but I don't have the bandwidth to tackle making more changes today/this weekend.  Our IDD freeze has technically passed, so unless you are able to make the changes to complete the checklist, I think this will have to wait.\r\n\r\nI'll mark it for version \"9.0\" (which may be 8.9), so it can drop in as soon as we've released 8.8.  If you do get around to doing @mjwitte's checklist of requests today, and it looks good, just ping me and I can try to get this into 8.8.  If you want to simply hand it off and let us finish it up, that's cool too!\r\n\r\nHowever it works out, thanks for all your contributions!"}],"action":{"name":"View Pull Request","url":"https://github.com/NREL/EnergyPlus/pull/6169#issuecomment-323518389"}}}

jmarrec commented:
Per @rraustad's request, here's a before/after.

Take `WaterHeaterDHWPlantLoop.idf`, and put this:

```
! m^3 is default anyways, but currently you need to Harcode UserDefined and set to 1 for Energy Conversion Factor,
! otherwise it goes kWh and doesn't report cost

UtilityCost:Tariff,
    Water Tariff,              !- Name
    Water:Facility,          !- Output Meter Name
    ,                        !- Conversion Factor Choice
    ,                       !- Energy Conversion Factor
    ,                        !- Demand Conversion Factor
    ,                        !- Time of Use Period Schedule Name
    ,                        !- Season Schedule Name
    ,                        !- Month Schedule Name
    ,                        !- Demand Window Length
    1.77,                    !- Monthly Charge or Variable Name
    ,                        !- Minimum Monthly Charge or Variable Name
    ,                        !- Real Time Pricing Charge Schedule Name
    ,                        !- Customer Baseline Load Schedule Name
    ,                        !- Group Name
    BuyFromUtility;          !- Buy Or Sell
    
UtilityCost:Charge:Simple,
    FlatWaterChargePerm3,   !- Name
    Water Tariff,             !- Tariff Name
    totalEnergy,            !- Source Variable
    Annual,                 !- Season
    EnergyCharges,          !- Category Variable Name
    3.3076;                 !- Cost per Unit Value or Variable Name
```

## Before

The resulting Tariff Section shows no cost aside from the Monthly charge.

![screen shot 2017-09-01 at 10 52 35 am](https://user-images.githubusercontent.com/5479063/29962497-b8c917ae-8f03-11e7-831f-7c6a5329d234.png)


## After:

There is a charge now. (the sim is run for 72 hours in January, from Jan 1 to 3)

![screen shot 2017-09-01 at 10 53 56 am](https://user-images.githubusercontent.com/5479063/29962557-eab8a112-8f03-11e7-84b0-ce9ca02b603d.png)

Here's an extract from the eplusout.err:

```
   ** Warning ** GetInputEconomicsTariff: UtilityCost:Tariff="WATER TARIFF" invalid data
   **   ~~~   ** Conversion Factor Choice="", Defaulting to m^3 (Water resource detected).
```

jmarrec committed:
Implemented a "smart" assignment of conversion factors based on a resource type in GetInputEconomicsTariff.

jmarrec committed:
Update documentation (untested, don't have tex)

jmarrec commented:
I'm only noticing now that E+ assumes that 1 CCF = 1 therm, when in reality it differs (and the heat content of natural gas varies from region to region and year after year). According to the [EIA](https://www.eia.gov/tools/faqs/faq.php?id=45&t=8), in 2016 1 CCF = 1.037 therms.

jmarrec commented:
I've rebased on the current develop, moved all determination of resource type and associated conversion factors in `GetInputEconomicsTariff`. Left placeholders in the IDD for now (since frozen) to add m3, gal and kgal.

I've updated the latex documentation to explicitly say what is allowed, and what will be the conversion factor, for each resource type. For example if you put "m3" for an electric meter, it will issue a warning and reset it to "kWh". A special warning is issued if electric and therm is picked, to let the user know it is unusual (but it isn't been reset to kWh)

Posting here as an image for reference:

![screen shot 2017-09-01 at 1 57 59 pm](https://user-images.githubusercontent.com/5479063/29968800-9c0369e8-8f1d-11e7-9de7-881229c1c41f.png)

jmarrec commented:
Eplus compiles just fine now, but I'm having an issue that I'm not sure about:

When I run the simulation, it crashes. Eplusout.err only has

     ** Warning ** Version: in IDF="8.7" not the same as expected="8.8"

And the terminal output is:

```sh
EnergyPlus Starting
EnergyPlus, Version 8.8.0-7a7f823b23, YMD=2017.09.01 13:48
Processing Data Dictionary
Processing Input File
Initializing Simulation
Reporting Surfaces
Beginning Primary Simulation
Initializing New Environment Parameters
Warming up {1}
Starting Simulation at 01/01 for Chicago Ohare Intl Ap IL USA TMY3 WMO#=725300

Assertion failed: (contains( i )), function operator(), file /Users/julien/Software/Others/EnergyPlus/third_party/ObjexxFCL/src/ObjexxFCL/Array1.hh, line 1164.
./run_custom.sh: line 9: 13534 Abort trap: 6
```

I have used a poor man's hack of using `std::cout` statement and `GetInputEconomicsTariff` does run just fine.

Note that I didn't start with a "clean" build folder, but I was building E+ yesterday on develop and it seemed to work better.

Also note that I didn't use IDFversionUpdater (requires building the package apparently?) to update the 8.7 file so maybe that just it? See attached idf (rename extension to idf)

[WaterHeaterDHWPlantLoop-Mod.txt](https://github.com/NREL/EnergyPlus/files/1270135/WaterHeaterDHWPlantLoop-Mod.txt)

jmarrec commented:
Ok I managed to build the Transition and update the IDF to 8.8. Works now, see attached.
[WaterHeaterDHWPlantLoop-Mod.txt](https://github.com/NREL/EnergyPlus/files/1270409/WaterHeaterDHWPlantLoop-Mod.txt)

rraustad commented:
Your new unit test is failing now.

mjwitte commented:
@jmarrec Thanks for going the extra mile on this.  The assertion failure indicates an array bounds problem - either the array isn't allocated yet, or it's being passed a 0 index.  If you run the file in the debugger, you can find out what line the failure is on by looking at the call stack when it fails.  That's more for future reference, since you're past that now.  But it is a bit troubling that the v8.7 idf crashed instead of fataling out properly - I wonder if it does that with develop or just on this branch.

jmarrec committed:
Added more Gtest

jmarrec committed:
Updated Gtest

jmarrec commented:
@rraustad: Unit tests are fine, I was just updating (and expanding them) in the meantime. I have declared new ints, including `conversionM3` and gtest was still expecting `conversionUSERDEF`.  They all pass now.

@mjwitte: I am **very** interested in knowing how I can debug E+ more easily including seeing stack trace, and possibly put breakpoints during execution.
Thing is, I'm on mac. I've seen a couple of entries in the wiki about debugging, but most refer to Visual Studio (windows), except [Debugging STL](https://github.com/NREL/EnergyPlus/wiki/Debugging-STL-objects-in-GDB) which I did not follow *at all*. 
And yes, I'm also troubled by the fact that it fataled out too. (The 8.7 IDF that crashed is uploaded a few posts above)

mjwitte commented:
I'm not familiar with the Mac development tools, but they should work the same way - put the input files in.idf and in.epw in the Build\Products folder, and then start EnergyPlus with the debugger.  Perhaps @RKStrand can advise on debugging on Mac - and then perhaps add a Wiki section for that.

jmarrec commented:
I'm building E+ in debug right now, I'll try LLDB to see how that works. I'll figure it out.

mbadams5 commented:
@jmarrec When I need to debug on my Mac, I typically use [CLion](https://www.jetbrains.com/clion/) which works very well. I have also used XCode (and still do sometimes) to debug. It is a little more involved to get working with an externally compiled program but possible. We can chat on slack if you want.

jmarrec commented:
So using LLDB, I can see that the E+ 8.7 file actually crashes in the LEED reporting section [here](https://github.com/jmarrec/EnergyPlus/blob/f47bea4ca814de3e9c6e1a3edfed8d771e47a127/src/EnergyPlus/EconomicTariff.cc#L4068)

I did leave a [TODO](https://github.com/jmarrec/EnergyPlus/blob/f47bea4ca814de3e9c6e1a3edfed8d771e47a127/src/EnergyPlus/EconomicTariff.cc#L114) because I didn't know if I needed to extend this, I guess I do.

Weird that it doesn't crash on 8.8.0 though

jmarrec committed:
Extending the convEneStrings to avoid error in LEED section, fyi @JasonGlazer left a note for you.

nrel-bot-3 commented:
@Myoldmopar @jmarrec @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @jmarrec @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @jmarrec @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @jmarrec @lgentile it has been 14 days since this pull request was last updated.

nrel-bot-3 commented:
@Myoldmopar @jmarrec @lgentile it has been 14 days since this pull request was last updated.

JasonGlazer commented:
@jmarrec I was going to start doing a code review. Is this completely done?

JasonGlazer commented on src/EnergyPlus/EconomicTariff.cc:
```diff
@@ -448,46 +464,211 @@ namespace EconomicTariff {
+				} else if ( SameString( cAlphaArgs( 3 ), "M3" ) ) {
+					tariff( iInObj ).convChoice = conversionM3;
+					tariff( iInObj ).energyConv = 1.0;
+					tariff( iInObj ).demandConv = 1.0;
```
The demand conversion factor should probably be based on M3/hour = 3600.0

JasonGlazer commented on src/EnergyPlus/EconomicTariff.cc:
```diff
@@ -448,46 +464,211 @@ namespace EconomicTariff {
+				} else if ( SameString( cAlphaArgs( 3 ), "CCF" ) ) {
+					tariff( iInObj ).convChoice = conversionCCF;
+					tariff( iInObj ).energyConv = 0.35314666721488586;
+					tariff( iInObj ).demandConv = 0.35314666721488586;
```
Change here and remaining demand conversions as well.

JasonGlazer commented on idd/Energy+.idd.in:
```diff
@@ -89752,6 +89752,9 @@ UtilityCost:Tariff,
         \key kBtu
         \key MCF
         \key CCF
+!        \key m3
```
Please put them into the IDD now.

JasonGlazer commented on src/EnergyPlus/EconomicTariff.cc:
```diff
@@ -107,9 +107,13 @@ namespace EconomicTariff {
 
-	Array1D_string const convEneStrings( {0,7}, { "", "kWh", "Therm", "MMBtu", "MJ", "kBtu", "MCF", "CCF" } );
-	Array1D_string const convDemStrings( {0,7}, { "", "kW", "Therm", "MMBtu", "MJ", "kBtu", "MCF", "CCF" } );
+	// TODO: @JasonGlazer, I extended this to include the water stuff, not sure if right for LEED or not.
```
I think this should be ok for the existing code for LEED reporting

JasonGlazer commented on src/EnergyPlus/EconomicTariff.cc:
```diff
@@ -448,46 +464,211 @@ namespace EconomicTariff {
+				}
+
+			// If it's an electric meter
+			// TODO For reviewer: do we want to actually limit the accepted choices?
```
Issuing warning when weird choices are made is probably the best option

JasonGlazer commented on src/EnergyPlus/EconomicTariff.hh:
```diff
@@ -314,6 +323,8 @@ namespace EconomicTariff {
 		int reportMeterIndx; // index of the report meter
 		int kindElectricMtr; // kind of electric meter - see enumerated list above, 0 is not electric
+		int kindWaterMtr; // kinf of water meter - 0 (default) is not water, 1 is water
+		int kindGasMtr; // kinf of gas meter - 0 (default) is not gas, 1 is gas
```
What is the logic for just have this for electric, water and gas and not other types of sources?

jmarrec replied:
Because you don't have the same conversion factors between Water and Gas while you can have the same underlying base unit of 'm3'. I may want to convert a gas meter from Joules to for eg CCF (100 ft³) of gas, while I may want to convert a water meter from m3 to CCF.

jmarrec replied:
I should have explained the logic clearly enough in the cc file using comments, see for eg [here](https://github.com/jmarrec/EnergyPlus/blob/HandleWaterUtilityCostTariffs/src/EnergyPlus/EconomicTariff.cc#L633)

I implemented the kindWAterMtr and kingGasMtr (kindElectricMtr already existing) because there was something to be done there. For the other resources, I just assumed default, for a few reasons:

* I didn't care too much about other much less used resources to be honest.
* For a lot of them, you cannot infer the heat content which is necessary to convert to volumetric units. For Natural Gas, E+ assumed 1 therm = 1 CCF, which is an approximation but I'm ok with it. What about "FuelOil#1" and "FuelOil#2" (Is that actually grade 1 and 2 (=Bunker A) of the 6-scale classification used in north america?!)? Worse: OtherFuel1 and OtherFuel2?

JasonGlazer commented on tst/EnergyPlus/unit/EconomicTariff.unit.cc:
```diff
@@ -225,6 +225,184 @@ TEST_F( EnergyPlusFixture, EconomicTariff_GetInput_Test)
+
+
+
+
```
Thanks for the unit tests

JasonGlazer reviewed (commented):
Thanks for the effort. Can you please merge from develop it has changed a lot

jmarrec committed:
Merge branch 'develop' into HandleWaterUtilityCostTariffs

jmarrec committed:
Actual idd change + adressing jason's comments

jmarrec commented:
@JasonGlazer merged develop in my branch + addressed your comments.

jmarrec commented:
I have to modify the unit tests now that I changed the demand conversion factor. I'll do that tomorrow, have to run now.

jmarrec committed:
Fix gtest

JasonGlazer commented:
@jmarrec I reviewed your code and ran some files and it generally looks good. I just have two comments:

- Mike Witte's original suggestion: "Edit the IDD notes to clarify how the default will be selected when the conversion factor is blank." still needs to be done

- The General table in the Tariff report should show the units. Using the modified WaterHeaterDHWPlantLoop.idf file it ends up blank.

Thanks. I still need to look at the revised PDF.

JasonGlazer commented:
@jmarrec I just took a look at the PDF. I think the energy meter text should come before the water meter since it is probably more common for most users. Also demand conversion factor for water meters should be in the table. In general, the paragraph breaks seem weird in the PDF. Some headings that should be bold are not shown that way. One of the tables that I think is supposed to be about gas meters is labeled electric meters. The note about heat content should probably follow the gas conversion table. Have you been able to generate the PDF yourself? I'm having problems with that this morning. The CI does create the PDFs in the  x86_64-Linux-Ubuntu-14.04-gcc-4.8 build.

jmarrec commented:
I mentioned that a while ago either in a comment or commit message but I did edits to latex without a way to generate latex pdf, I didn’t have 2-3 GB of drive space to spare at that time.
I’m really running low on *free* time to polish this, is it absolutely necessary that I do these last few things (honest question)?

JasonGlazer committed:
Minor fixes to IOref and IDD.

JasonGlazer commented:
@jmarrec I made some small fixes to the IOref and IDD that should address the two suggestions I had made a few day ago. Let me know if you agree. The Linux build from the CI should (eventually) regenerate the IOref pdf so that you can see the impact of the changes.

jmarrec committed:
Small fixes

JasonGlazer commented:
@Myoldmopar and @mjwitte  The x86_64-Linux-Ubuntu-14.04-custom_check build is failing with the following message for each source file:

[license-check] License year is incorrect

Is this because it is expecting 2018. If so, I assume that should be getting fixed in a different pull request.

mjwitte commented:
Yes, see #6425 which should be merged in soon.

JasonGlazer committed:
Merge branch 'develop' into HandleWaterUtilityCostTariffs

jmarrec committed:
Merge latest upstream/develop in

jmarrec commented:
@JasonGlazer here, I pulled the latest NREL/develop into this branch.

FYI, in the future, if you want do that yourself (or add any new commits for that matter), technically you - or any other member with write access - can just checkout my branch on your computer (that requires adding another remote, see [here](https://stackoverflow.com/questions/9153598/how-do-i-fetch-a-branch-on-someone-elses-fork-on-github)), pull the latest NREL/develop into it, and push back to my branch.

I have indeed checked the "Allow edits from maintainer box" (I set this by default):

> If checked, users with write access to NREL/EnergyPlus can add new commits to your HandleWaterUtilityCostTariffs branch. You can always change this setting later.

JasonGlazer committed:
Merge branch 'develop' into HandleWaterUtilityCostTariffs

JasonGlazer committed:
Merge branch 'HandleWaterUtilityCostTariffs' of https://github.com/jmarrec/EnergyPlus into HandleWaterUtilityCostTariffs

JasonGlazer commented:
@jmarrec thanks again for adding this feature.

jmarrec commented:
You're welcome. Thanks for merging.
